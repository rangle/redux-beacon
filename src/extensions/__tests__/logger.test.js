const { logger } = require('../logger');
const { makeConsoleMock } = require('consolemock');

beforeEach(() => {
  /* eslint-disable no-global-assign */
  console = makeConsoleMock();
});

const makeAction = () => ({ type: 'ROUTE_CHANGED', payload: '/home' });

/* eslint-disable no-console */
describe('logger()', () => {
  it('logs actions and any resulting analytics event', () => {
    const events = [{ hitType: 'pageview', page: '/home' }];
    const action = makeAction();
    logger(events, action);
    expect(console.printHistory()).toMatchSnapshot();
  });

  it('logs actions and any resulting analytics events', () => {
    const events = [
      { hitType: 'pageview', page: '/' },
      { hitType: 'pageview', page: '/home' },
    ];
    const action = makeAction();
    logger(events, action);
    expect(console.printHistory()).toMatchSnapshot();
  });

  it('logs events that are saved offline with a helpful message', () => {
    const events = [{ hitType: 'pageview', page: '/' }];
    const action = makeAction();
    logger(events, action, {}, true);
    expect(console.printHistory()).toMatchSnapshot();
  });

  it('logs events that were saved offline with a helpful message', () => {
    const events = [{ hitType: 'pageview', page: '/' }];
    const action = makeAction();
    logger(events, action, {}, false, true);
    expect(console.printHistory()).toMatchSnapshot();
  });

  it('logs nothing when there are no events', () => {
    const events = [];
    const action = makeAction();
    logger(events, action);
    expect(console.printHistory()).toMatchSnapshot();
  });
});
