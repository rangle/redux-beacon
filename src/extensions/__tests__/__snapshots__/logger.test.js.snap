// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logger() logs actions and any resulting analytics event 1`] = `
"
GROUP %c Analytics events @ 19:31:37.006 ROUTE_CHANGED, color: #1da1f2; font-weight: bold;
  LOG %c Action:, color: #93748A; font-weight: bold;, {type: 'ROUTE_CHANGED', payload: '/home'}
  LOG %c Event:, color: #93748A; font-weight: bold;, {hitType: 'pageview', page: '/home'}"
`;

exports[`logger() logs actions and any resulting analytics events 1`] = `
"
GROUP %c Analytics events @ 19:31:37.006 ROUTE_CHANGED, color: #1da1f2; font-weight: bold;
  LOG %c Action:, color: #93748A; font-weight: bold;, {type: 'ROUTE_CHANGED', payload: '/home'}
  GROUP %c Events:, color: #93748A; font-weight: bold;
    LOG %c (1), color: #93748A; font-weight: bold;, {hitType: 'pageview', page: '/'}
    LOG %c (2), color: #93748A; font-weight: bold;, {hitType: 'pageview', page: '/home'}"
`;

exports[`logger() logs events that are saved offline with a helpful message 1`] = `
"
GROUP %c Analytics events @ 19:31:37.006 ROUTE_CHANGED (saved offline), color: #D63230; font-weight: bold;
  LOG %c Action:, color: #93748A; font-weight: bold;, {type: 'ROUTE_CHANGED', payload: '/home'}
  LOG %c Event:, color: #93748A; font-weight: bold;, {hitType: 'pageview', page: '/'}"
`;

exports[`logger() logs events that were saved offline with a helpful message 1`] = `
"
GROUP %c Analytics events @ 19:31:37.006  (was offline), color: #F39237; font-weight: bold;
  LOG %c Event:, color: #93748A; font-weight: bold;, {hitType: 'pageview', page: '/'}"
`;

exports[`logger() logs nothing when there are no events 1`] = `""`;
